### ISSUE TITLE
`knownLength` causes socket hang up error

### ISSUE THREAD

**Issue by kevinprotoss** [form-data/form-data#154]
Hi,

I met across socket hang up error always if I set `knownLength` incorrectly, e.g., `knownLength` is 1, or smaller than actual length of file content.
If I removed `knownLength` option, it works.

Is this option be used for open file stream as well?

Bg,
Kevin

**Comment by alexindigo** [comment]
`knownLength` is used for data of "known" length, when you don't need/want FormData try to get/guess length of the content.

for regular filestreams FormData does `fs.stat` call to determine file size: https://github.com/form-data/form-data/blob/master/lib/form_data.js#L114

**Comment by kevinprotoss** [comment]
Yes, I understand. However, if I gave the knownLegnth but incorrect, e.g., smaller than actual length, what will happened? Why I always get such an error. It looks like the file stream is interrupted .
